src/windows/find_tools.rs

```diff
@@ -199,6 +203,31 @@ mod impl_ {
         include: Vec<PathBuf>,
     }
 
+    fn is_amd64_emulation_supported() -> bool {
+        // In theory we should free the library handle, but since kernel32.dll is loaded in every
+        // process and we're a short-lived process, leaking the handles doesn't matter.
+        let kernel32 = unsafe { LoadLibraryA(b"kernel32.dll\0".as_ptr() as _) };
+        // GetMachineTypeAttributes is only available on Win11 22000+.
+        if let Some(get_machine_type_attributes) =
+            unsafe { GetProcAddress(kernel32, b"GetMachineTypeAttributes\0".as_ptr() as _) }
+        {
+            let mut attributes = Default::default();
+            let get_machine_type_attributes: unsafe extern "system" fn(
+                u16,
+                *mut MACHINE_ATTRIBUTES,
+            ) -> HRESULT = unsafe { std::mem::transmute_copy(&get_machine_type_attributes) };
```

Comment:
Is it necessary to use `transmute` here?

Reply:
BTW, I'd like some comment explaining why these unsafe block is safe.

Reply:
Unfortunately, yes, `GetProcAddress` returns a `unsafe fn() -> isize`, so we need to transmute it into the correct function type.

Ideally, we'd have windows-sys generate the type that we transmute to, so that we can guarantee that it is correct, but I'm not sure how to do that without it being linked in as a dependency.

Reply:
Maybe we need to update the windows generator.

There's a new version but it removes all useful APIs and becomes a binary crate.

Not sure what's the way forward.

Reply:
Ok, I cleaned this up
* Updated to the latest `windows_bindgen`.
* Created a new list where we remove the `#[link]` attribute, which allows me to generate the definition for `GetMachineTypeAttributes` without statically linking it.

Reply:
That... didn't work.

It looks like any use of `GetMachineTypeAttributes` outside of a `const` scope will cause it to be linked into the binary. So, I've gone back to the hand-written definition BUT I've added a `const` block to make sure that the hand-written signature and the generated one match.